crates/full-node/sov-sequencer/src/preferred/db/mod.rs

```diff
@@ -387,34 +388,39 @@ impl PreferredSequencerDb {
         storage_path: &Path,
         postgres_connection_string: &Option<String>,
     ) -> anyhow::Result<(Self, bool)> {
-        let is_replica = is_replica.unwrap_or(false);
-        if is_replica {
-            return Ok((
+        if let Some(postgres_connection_string) = &postgres_connection_string {
+            let postgres_backend = PostgresBackend::connect(postgres_connection_string).await?;
+
+            match is_replica {
+                Some(true) => Ok((
+                    Self {
+                        backend: None,
+                        shutdown_sender: shutdown_sender.clone(),
+                    },
+                    true,
+                )),
+                Some(false) | None => {
```

Comment:
We handle is_replica = Some(false) and is_replica = None the same way for backward compatibility. This behavior will change in upcoming PRs, where None will indicate that the database determines the leader.